wordpress-objects.post.php

```diff
@@ -248,24 +248,82 @@ public function add_comment( $comment_text, $user_id ) {
 	 * @return array List of post tags.
 	 */
 	public function get_terms( $taxonomy, $args = array() ) {
-		return wp_get_post_terms( $this->get_id(), $taxonomy, $args );
+		return array_map(
+			function( $term ) use ( $taxonomy ) {
+				try {
+					return Term::get_by_slug( $term['slug'], $taxonomy );
+				} catch ( Exception $e ) {
+					return null;
+				}
+			},
+			wp_get_post_terms( $this->get_id(), $taxonomy, $args )
+		);
 	}
 
 	/**
-	 * Set the terms for a post.
+	 * Add terms to a post.
 	 *
-	 * @param string $tags     Optional. The tags to set for the post, separated by commas. Default empty.
-	 * @param string $taxonomy Optional. Taxonomy name. Default 'post_tag'.
-	 * @param bool   $append   Optional. If true, don't delete existing tags, just add on. If false,
-	 *                         replace the tags with the new tags. Default false.
-	 * @return mixed Array of affected term IDs. WP_Error or false on failure.
+	 * @param array $terms Term objects
 	 */
-	public function set_terms( $tags, $taxonomy, $append = false ) {
-		return wp_set_post_terms( $this->get_id(), $tags, $taxonomy, $append );
+	public function add_terms( $terms ) {
```

Comment:
@mattheu This should probably call `wp_set_object_terms` directly as you can pass all at once which I think will only cause one insert query.